src/sync/chase_lev.rs

```diff
@@ -158,69 +155,93 @@ impl<T> Deque<T> {
     unsafe fn push(&self, data: T) {
         let guard = epoch::pin();
 
-        let mut b = self.bottom.load(SeqCst);
-        let t = self.top.load(SeqCst);
-        let mut a = self.array.load(SeqCst, &guard).unwrap();
-        let size = b - t;
+        let mut b = self.bottom.load(Relaxed);
+        let t = self.top.load(Acquire);
+        let mut a = self.array.load(Relaxed, &guard).unwrap();
+        
+        let size = b.wrapping_sub(t);
```

Comment:
How come all the arithmetic operations were changed to wrapping? In general I would expect overflows here to indeed be fatal flaws that should be accounted for via different means elsewhere.